crates/starknet-types-core/src/chain_id/alloc_impls.rs

```diff
@@ -0,0 +1,191 @@
+#[cfg(not(feature = "std"))]
+pub extern crate alloc;
+#[cfg(not(feature = "std"))]
+use alloc::string::{String, ToString};
+
+use crate::short_string;
+use crate::short_string::ShortString;
+
+use super::{ChainId, SN_MAIN_STR, SN_SEPOLIA_STR};
+
+impl From<ChainId> for ShortString {
+    fn from(value: ChainId) -> Self {
+        match value {
+            ChainId::Mainnet => short_string!("SN_MAIN"),
+            ChainId::Sepolia => short_string!("SN_SEPOLIA"),
+            #[cfg(feature = "devnet")]
+            ChainId::Devnet(ss) => ss,
+        }
+    }
+}
+
+#[cfg(feature = "devnet")]
+impl From<ShortString> for ChainId {
+    fn from(value: ShortString) -> Self {
+        if value.as_ref() == SN_MAIN_STR {
+            ChainId::Mainnet
+        } else if value.as_ref() == SN_SEPOLIA_STR {
+            ChainId::Sepolia
+        } else {
+            ChainId::Devnet(value)
+        }
```

Comment:
```suggestion
        match value.as_ref() {
            SN_MAIN_STR => ChainId::Mainnet,
            SN_SEPOLIA_STR => ChainId::Sepolia,
            _ => ChainId::Devnet(value),
        }
```

Also, why does conversion from `ChainId` to `ShortString` is possible but not the other way around if feature `devnet` is disabled?

Reply:
Yep.
with devnet:
- any shortstring is a valid devnet name (we don't enforce it has to start with the `SN_` prefix, maybe we should, but I don't have strong opinion about that)
- TryFrom is automatically implemented as infaiilbe and will resolve to the From impl

without devnet:
- only two known values are valid `SN_MAIN_STR` and `SN_SEPOLIA_STR`, all other will result in an error
- therefor `From` is not implemented, only TryFrom